File: 03-thread-inseguro/bad.rb

@counter = 0

5.times.map do
  Thread.new do
    temp = @counter
    print "." if ARGV.length > 0
    temp = temp + 1
    @counter = temp
  end
end.each(&:join)

puts "El resultado siempre debería ser 5, pero fue #{@counter}"

File: 03-thread-inseguro/bad.py
import sys
import threading

counter = 0

def bad_done():
    global counter
    temp = counter
    if len(sys.argv) > 1:
        sys.stdout.write(".")
    temp = temp + 1
    counter = temp

for _ in range(5):
    threading.Thread(target=bad_done).start()

print("El resultado siempre debería ser 5, pero fue ", counter)
